metron-deployment/vagrant/KERBEROS_SETUP.md

```diff
@@ -0,0 +1,238 @@
+```
+
+14. Add metron user to the Kafka cluster ACL
+```
+/usr/hdp/current/kafka-broker/bin/kafka-acls.sh --authorizer kafka.security.auth.SimpleAclAuthorizer --authorizer-properties zookeeper.connect=node1:2181 --add --allow-principal User:${KERB_USER} --cluster kafka-cluster
+```
+
+15. We also need to grant permissions to the HBase tables. Kinit as the hbase user and add ACLs for metron.
+```
+kinit -kt /etc/security/keytabs/hbase.headless.keytab hbase-metron_cluster@EXAMPLE.COM
+echo "grant 'metron', 'RW', 'threatintel'" | hbase shell
+echo "grant 'metron', 'RW', enrichment" | hbase shell
+```
+
+16. Create a “.storm” directory in the metron user’s home directory and switch to that directory.
+```
+su - metron
+mkdir .storm
+cd .storm
+```
+
+17. Create a custom client jaas file. This should look identical to the Storm client jaas file located in /etc/storm/conf/client_jaas.conf except for the addition of a Client stanza. The Client stanza is used for Zookeeper. All quotes and semicolons are necessary.
+```
+[metron@node1 .storm]$ cat client_jaas.conf
+StormClient {
+   com.sun.security.auth.module.Krb5LoginModule required
+   useTicketCache=true
+   renewTicket=true
+   serviceName="nimbus";
+};
+Client {
+   com.sun.security.auth.module.Krb5LoginModule required
+   useTicketCache=true
+   renewTicket=true
+   serviceName="zookeeper";
+};
+KafkaClient {
+   com.sun.security.auth.module.Krb5LoginModule required
+   useTicketCache=true
+   renewTicket=true
+   serviceName="kafka";
+};
+```
+
+18. Create a storm.yaml with jaas file info.
+```
+[metron@node1 .storm]$ cat storm.yaml
+nimbus.seeds : ['node1']
+java.security.auth.login.config : '/home/metron/.storm/client_jaas.conf'
+storm.thrift.transport : 'org.apache.storm.security.auth.kerberos.KerberosSaslTransportPlugin'
+```
+
+19. Create an auxiliary storm configuration json file in the metron user’s home directory. Note the login config option in the file points to our custom client_jaas.conf.
+```
+cd /home/metron
+[metron@node1 ~]$ cat storm-config.json
+{
+  "topology.worker.childopts" : "-Djava.security.auth.login.config=/home/metron/.storm/client_jaas.conf"
+}
+```
+
+20. Setup enrichment and indexing.
+    1. Modify enrichment.properties - `${METRON_HOME}/config/enrichment.properties`
+```
+kafka.security.protocol=PLAINTEXTSASL
+topology.worker.childopts=-Djava.security.auth.login.config=/home/metron/.storm/client_jaas.conf
+```
+
+    2. Modify elasticsearch.properties - `${METRON_HOME}/config/elasticsearch.properties`
+```
+kafka.security.protocol=PLAINTEXTSASL
+topology.worker.childopts=-Djava.security.auth.login.config=/home/metron/.storm/client_jaas.conf
+```
+
+21. Restart the parser topologies. Be sure to pass in the new parameter, “-ksp” or “--kafka_security_protocol.” Run this from the metron home directory.
+```
+for parser in bro snort; do ${METRON_HOME}/bin/start_parser_topology.sh -z node1:2181 -s ${parser} -ksp PLAINTEXTSASL -e storm-config.json; done
```

Comment:
Did a round of troubleshooting with @mmiklavc - thank you, Mike!

- Need to do `kinit` as metron user for the above error to go away
- Further, the `kafka_security_protocol` needs to be changed as `SASL_PLAINTEXT` instead of `PLAINTEXTSASL` in order to avoid `worker died` errors in the storm topology